src/bootstrap/test.rs

```diff
@@ -483,6 +485,10 @@ impl Step for Miri {
             target: self.host,
             extra_features: Vec::new(),
         });
+        // The stdlib we need might be at a different stage. And just asking for the
+        // sysroot does not seem to populate it, so we do that first.
+        builder.ensure(compile::Std::new(compiler_std, host));
+        let sysroot = builder.sysroot(compiler_std);
```

Comment:
Is it expected that I need to call `ensure` and `sysroot`, i.e., that calling `sysroot` does not implicitly `ensure` anything?

Reply:
It ensures the directory exists. It doesn't build the standard library, a) because it doesn't know what target you want and b) because this is also called when building Std so we can copy the newly built artifacts into the sysroot.